module-audio/Audio/test/unittest_config_utils.cpp

```diff
@@ -0,0 +1,61 @@
+        REQUIRE(config.flags == 1);
+        REQUIRE(config.inputGain == 2.0);
+        REQUIRE(config.outputVolume == 1.0);
+        REQUIRE(config.inputPath == audio::codec::InputPath::None);
+        REQUIRE(config.outputPath == audio::codec::OutputPath::None);
+
+        audio::equalizer::Equalizer filterCoefficients = {
+            qfilter_CalculateCoeffs(audio::equalizer::FilterType::None, 1000.2f, 44100, 0.7f, 10),
+            qfilter_CalculateCoeffs(audio::equalizer::FilterType::HighPass, 2000.0f, 8000, 1.7f, -10),
+            qfilter_CalculateCoeffs(audio::equalizer::FilterType::LowPass, 10000.0f, 44100, 0.75f, 2.5),
+            qfilter_CalculateCoeffs(audio::equalizer::FilterType::Notch, 2500.0f, 44100, 4.4f, 5.3),
+            qfilter_CalculateCoeffs(audio::equalizer::FilterType::BandPass, 1000, 44100, 0.7f, 10)};
+
+        REQUIRE(config.filterCoefficients.at(0).a1 == filterCoefficients.at(0).a1);
+        REQUIRE(config.filterCoefficients.at(0).a2 == filterCoefficients.at(0).a2);
+        REQUIRE(config.filterCoefficients.at(0).b0 == filterCoefficients.at(0).b0);
+        REQUIRE(config.filterCoefficients.at(0).b1 == filterCoefficients.at(0).b1);
+        REQUIRE(config.filterCoefficients.at(0).b2 == filterCoefficients.at(0).b2);
+
+        REQUIRE(config.filterCoefficients.at(1).a1 == filterCoefficients.at(1).a1);
+        REQUIRE(config.filterCoefficients.at(1).a2 == filterCoefficients.at(1).a2);
+        REQUIRE(config.filterCoefficients.at(1).b0 == filterCoefficients.at(1).b0);
+        REQUIRE(config.filterCoefficients.at(1).b1 == filterCoefficients.at(1).b1);
+        REQUIRE(config.filterCoefficients.at(1).b2 == filterCoefficients.at(1).b2);
+
+        REQUIRE(config.filterCoefficients.at(2).a1 == filterCoefficients.at(2).a1);
+        REQUIRE(config.filterCoefficients.at(2).a2 == filterCoefficients.at(2).a2);
+        REQUIRE(config.filterCoefficients.at(2).b0 == filterCoefficients.at(2).b0);
+        REQUIRE(config.filterCoefficients.at(2).b1 == filterCoefficients.at(2).b1);
+        REQUIRE(config.filterCoefficients.at(2).b2 == filterCoefficients.at(2).b2);
+
+        REQUIRE(config.filterCoefficients.at(3).a1 == filterCoefficients.at(3).a1);
+        REQUIRE(config.filterCoefficients.at(3).a2 == filterCoefficients.at(3).a2);
+        REQUIRE(config.filterCoefficients.at(3).b0 == filterCoefficients.at(3).b0);
+        REQUIRE(config.filterCoefficients.at(3).b1 == filterCoefficients.at(3).b1);
+        REQUIRE(config.filterCoefficients.at(3).b2 == filterCoefficients.at(3).b2);
+
+        REQUIRE(config.filterCoefficients.at(4).a1 == filterCoefficients.at(4).a1);
+        REQUIRE(config.filterCoefficients.at(4).a2 == filterCoefficients.at(4).a2);
+        REQUIRE(config.filterCoefficients.at(4).b0 == filterCoefficients.at(4).b0);
+        REQUIRE(config.filterCoefficients.at(4).b1 == filterCoefficients.at(4).b1);
+        REQUIRE(config.filterCoefficients.at(4).b2 == filterCoefficients.at(4).b2);
```

Comment:
use the for loop Luke? :) what exactly do we test here? :)

Reply:
my mind was blackened during writing those checks :D sure - will change to for loop
It is testing if the parameters read from the file are the same as calculated with the same data provided, thus it checks if the data was parsed correctly. Any mismatch would result in a different coefficients

Reply:
tbh you can add `operator==` and just compare the `QFilterCoefficients` as you are actually doing that, don't you? :)

Reply:
yep - I did the same in Bluetooth settings :thinking:

Reply:
fixed :D